extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -588,27 +589,27 @@ private Optional<SqlStatementResult> getStatementStatus(
     MSQControllerTask msqControllerTask = getMSQControllerTaskAndCheckPermission(queryId, authenticationResult, forAction);
     SqlStatementState sqlStatementState = SqlStatementResourceHelper.getSqlStatementState(statusPlus);
 
-    Supplier<Optional<MSQTaskReportPayload>> msqTaskReportPayloadSupplier = () -> {
+    Supplier<MSQTaskReportPayload> msqTaskReportPayloadSupplier = Suppliers.memoize(() -> {
```

Comment:
Why do we require a supplier? We are using the result directly, and since we don't call the supplier multiple times now, it is easier and equivalent to remove the supplier.

Reply:
Following is a lot simpler than memoizing a supplier. Suppliers are usually beneficial when we need to compute the result lazily. In this method, the supplier is invoked instantaneously. 
```java
  MSQTaskReportPayload taskReportPayload = null
  if (detailed == true) {
   taskReportPayload = contactOverlord...
  }
```

Reply:
Have made the change.